From 8354c2e09127c21bf06685e961101eedc2d40574 Mon Sep 17 00:00:00 2001
From: Yongxin Liu <yongxin.liu@windriver.com>
Date: Thu, 13 Jun 2024 10:27:03 +0000
Subject: [PATCH 19/24] rdk/snr: replace vma->vm_flags direct modifications

kernel commit 1c71222e5f23 ("mm: replace vma->vm_flags direct
modifications with modifier calls").

Upstream-Status: Inappropriate [RDK is released in tarball only]

Signed-off-by: Yongxin Liu <yongxin.liu@windriver.com>
---
 klm/cpk-ae/ice_sw_ae_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/klm/cpk-ae/ice_sw_ae_main.c b/klm/cpk-ae/ice_sw_ae_main.c
index 9a350ee..df16b46 100644
--- a/klm/cpk-ae/ice_sw_ae_main.c
+++ b/klm/cpk-ae/ice_sw_ae_main.c
@@ -602,7 +602,7 @@ static int ice_sw_ae_mmap(struct file *f, struct vm_area_struct *vma)
 		per_proc_info = f->private_data;
 		ae_dev = per_proc_info->ae_dev;
 
-		vma->vm_flags |= VM_DONTEXPAND | VM_DONTDUMP;
+		vm_flags_set(vma, VM_DONTEXPAND | VM_DONTDUMP);
 
 		ret = remap_pfn_range(vma,
 				      vma->vm_start,
-- 
2.39.3

